Kitodo-Query-URL-Import/src/main/java/org/kitodo/queryurlimport/QueryURLImport.java

```diff
@@ -269,11 +269,10 @@ private DataRecord performQueryToRecord(DataImport dataImport, String queryURL,
         String prefix = Objects.nonNull(idPrefix) && !identifier.startsWith(idPrefix) ? idPrefix : "";
         String idParameter = SearchInterfaceType.OAI.equals(dataImport.getSearchInterfaceType()) ? OAI_IDENTIFIER
                 : dataImport.getIdParameter();
-        String queryParameter = idParameter + EQUALS_OPERAND + prefix + identifier;
         if (SearchInterfaceType.SRU.equals(interfaceType)) {
-            fullUrl += URLEncoder.encode(queryParameter, encoding);
+            fullUrl += URLEncoder.encode(idParameter + EQUALS_OPERAND + prefix + identifier, encoding);
         } else {
-            fullUrl += queryParameter;
+            fullUrl += URLEncoder.encode(idParameter, encoding) + EQUALS_OPERAND + URLEncoder.encode(prefix + identifier, encoding);
```

Comment:
It is probably not necessary to encode the parameter name as well. I am not sure but i could maybe also lead to problems server side? But i am not sure about that and have not tested it.

Reply:
I would agree that in most cases (including the issue in question) it would be sufficient to just encode the value.
As far as I know, however, it is technically possible or permissible to use these characters in the parameter name as well. Therefore, my suggestion was to also encode this as a precaution.

Reply:
I suppose it is highly unlikely that people will use parameter names which contain special charcters and standard names like "id" or "title" will not be changed on encoding. So normal search backend will not be affected by your change i suppose. People who are using parameter names with special characters can probably also handle them on the Backend. Therefor i agree with you.